builtin/logical/transit/path_encrypt.go

```diff
@@ -158,6 +175,145 @@ func (b *backend) pathEncryptWrite(
 		return logical.ErrorResponse("policy not found"), logical.ErrInvalidRequest
 	}
 
+	batchInputRaw := d.Get("batch").(string)
+	var batchInput []byte
+	if len(batchInputRaw) != 0 {
+		batchInput, err = base64.StdEncoding.DecodeString(batchInputRaw)
+		if err != nil {
+			return logical.ErrorResponse("failed to base64-decode batch"), logical.ErrInvalidRequest
+		}
+	}
+
+	if len(batchInput) != 0 {
```

Comment:
I think a cleaner implementation would be to first figure out if there is batch input; if not, pop the given ciphertext and context into a single batch item and run it through. Then at the end, if batch was not given, pull the values into the normal-style response.

It's a bit of data haggling back and forth, but it also means that the control flow ends in all cases at the end of the function, rather than a large `if` block that exits the function in the middle.